rskj-core/src/test/java/co/rsk/db/MutableTrieCacheTest.java

```diff
@@ -323,4 +324,79 @@ public void testStorageKeysDeletedAccount() {
         Assert.assertTrue(keys.contains(sk121));
         Assert.assertEquals(1, keys.size());
     }
+
+    @Test
+    public void testStoreValueOnTrieAndGetSize() {
+        MutableTrieImpl baseMutableTrie = new MutableTrieImpl(null, new Trie());
+
+        // First put some strings in the base
+
+        byte[] value = toBytes("alice");
+        byte[] key = toBytes("ALICE");
+        byte[] keyForCache = toBytes("ALICE2");
+
+        baseMutableTrie.put(key, value);
+        byte[] result = baseMutableTrie.get(key);
+        Assert.assertArrayEquals(value, result);
+
+        Uint24 valueLength = baseMutableTrie.getValueLength(key);
+        Assert.assertEquals(valueLength, new Uint24(value.length));
+
+        // Test the same in cache
+
+        MutableTrieCache mtCache = new MutableTrieCache(baseMutableTrie);
```

Comment:
I would move the `co.rsk.db.MutableTrieCache` tests to its own class. Or maybe we can parametrize the test receiving two `co.rsk.trie.MutableTrie` implementations, one `co.rsk.db.MutableTrieCache` and one `co.rsk.db.MutableTrieImpl`.

Reply:
> This looks great, I just made minor comments.
> I think `9373882647277a3273aae53ea2597f5281e60d11` can be fixed up into `6c1fa26dae6086916a02e92b8ee5bcfdda9cfca6`.
> Also, are the [test cases described in EIP-1052](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1052.md#test-cases) covered by the _integration tests_ suite?

Yes, you can check it [here](https://github.com/rsksmart/rskj-truffle-tests/pull/1/commits/bdfc52b6305ccd8c40f6f94d23cacee833158ba5), each test has which case is testing.

Reply:
About co.rsk.db.MutableTrieCache tests, i made some [changes](https://github.com/rsksmart/rskj/blob/2f8340b236b5508ee07875bf8cb8ba7488c560be/rskj-core/src/test/java/co/rsk/db/MutableTrieCacheTest.java#L406)